ipv8/REST/identity_endpoint.py

```diff
@@ -30,11 +30,11 @@ class IdentityEndpoint(BaseEndpoint):
     def __init__(self, middlewares=()):
         super(IdentityEndpoint, self).__init__(middlewares)
         self.communication_manager = None
-        self.app.on_shutdown.append(self.on_shutdown)
 
     def initialize(self, session):
         super(IdentityEndpoint, self).initialize(session)
         self.communication_manager = CommunicationManager(session)
+        self.app.on_shutdown.append(self.on_shutdown)
```

Comment:
This line is modifying the declared callback routes after their creation. This is not allowed, explaining the crash. This line change should be reverted.

Instead, in `on_shutdown` it should be checked `if self.communication_manager == None:` before `await self.communication_manager.shutdown()`.

Reply:
Yeah. I'm running with the thing disabled. I'll correct the fix.